compose-wireshark-plugin/src/main.rs

```diff
@@ -0,0 +1,187 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use anyhow::Context;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use convert_case::Case;
+use convert_case::Casing as _;
+use gateway_messages::MessageKind;
+use gateway_messages::MgsError;
+use gateway_messages::MgsRequest;
+use gateway_messages::MgsResponse;
+use gateway_messages::SpError;
+use gateway_messages::SpRequest;
+use gateway_messages::SpResponse;
+use std::fs;
+use std::io;
+use std::io::BufWriter;
+use std::io::Write as _;
+use strum::VariantNames;
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[clap(default_value = "wireshark")]
+    output_dir: Utf8PathBuf,
+}
+
+struct LuaWriter {
+    output_dir: Utf8PathBuf,
+    output_protofields: BufWriter<fs::File>,
+}
+
+impl LuaWriter {
+    fn new(output_dir: Utf8PathBuf) -> anyhow::Result<Self> {
+        fs::create_dir_all(&output_dir).with_context(|| {
+            format!("failed to create output dir {output_dir}")
+        })?;
+
+        let path = output_dir.join("protofields.lua");
+        println!("creating or overwriting {path}");
+        let output_protofields =
+            BufWriter::new(fs::File::create(&path).with_context(|| {
+                format!("failed to create output file {path}")
+            })?);
+
+        Ok(Self { output_dir, output_protofields })
+    }
+
+    fn write_protofields_preamble(&mut self) -> anyhow::Result<()> {
+        || -> io::Result<()> {
+            let f = &mut self.output_protofields;
+
+            writeln!(
+                f,
+                "-- This file is auto-generated. Any hand edits \
+                    will be overwritten!"
+            )?;
+            writeln!(f)?;
+            writeln!(f, "local M = {{}}")?;
+            writeln!(f)?;
+
+            Ok(())
+        }()
+        .with_context(|| {
+            format!("failed to write to {}/protofields.lua", self.output_dir)
+        })
```

Comment:
turbo nit: this shows up all over the place, maybe it could be stuck in a function or something?